library/src/image.cxx

```diff
@@ -21,25 +25,44 @@ class image::internals
   unsigned int Width = 0;
   unsigned int Height = 0;
   unsigned int Channels = 0;
+  unsigned int ChannelSize = 1;
 
-  void UpdateBufferSize() { this->Buffer.resize(this->Width * this->Height * this->Channels); }
+  void UpdateBufferSize()
+  {
+    this->Buffer.resize(this->Width * this->Height * this->Channels * this->ChannelSize);
+  }
 
   vtkSmartPointer<vtkImageImport> GetVTKImporter() const
   {
-    assert(this->Buffer.size() == this->Width * this->Height * this->Channels);
+    assert(this->Buffer.size() == this->Width * this->Height * this->Channels * this->ChannelSize);
 
     vtkNew<vtkImageImport> importer;
     importer->CopyImportVoidPointer(
       const_cast<unsigned char*>(this->Buffer.data()), static_cast<vtkIdType>(this->Buffer.size()));
     importer->SetNumberOfScalarComponents(this->Channels);
-    importer->SetDataScalarTypeToUnsignedChar();
+    switch (this->ChannelSize)
+    {
+      case 1:
+        importer->SetDataScalarType(VTK_UNSIGNED_CHAR);
+        break;
+      case 2:
+        importer->SetDataScalarType(VTK_UNSIGNED_SHORT);
+        break;
+      case 4:
+        importer->SetDataScalarType(VTK_FLOAT);
```

Comment:
naive question here. Is there any image format of channel size 4 that is not VTK_FLOAT, eg VTK_UNSIGNED_INT ?